packages/monitor-v2/src/monitor-polymarket/MonitorProposalsOrderBook.ts

```diff
@@ -54,7 +54,14 @@ export async function monitorTransactionsProposedOrderBook(
   const marketsWithEventData: PolymarketWithEventData[] = marketsWithAncillary
     .filter((market) => proposalEvents.find((event) => event.ancillaryData === market.ancillaryData))
     .map((market) => {
-      const event = proposalEvents.find((event) => event.ancillaryData === market.ancillaryData);
+      const events = proposalEvents.filter((event) => event.ancillaryData === market.ancillaryData);
+      // get last proposal event
+      const event = events.reduce((maxEvent, currentEvent) => {
+        if (currentEvent.proposalTimestamp > maxEvent.proposalTimestamp) {
+          return currentEvent;
+        }
+        return maxEvent;
+      });
```

Comment:
Instead of sorting by timestamp, should we sort by block/txn index/log index?

The reason I suggest this is that you could _technically have propose + dispute + propose in the same txn if an attacker was intending to fool the bot.

Reply:
Good point, done in latest commit